dbt/include/snowflake/macros/adapters.sql

```diff
@@ -150,21 +150,83 @@
 {% endmacro %}
 
 
-{% macro snowflake__list_relations_without_caching(schema_relation) %}
+{% macro snowflake__list_relations_without_caching(schema_relation, max_iter=10, max_results_per_iter=10000) %}
+
+  {%- set max_total_n = max_results_per_iter * max_iter -%}
+
   {%- set sql -%}
-    show terse objects in {{ schema_relation }}
+    show terse objects in {{ schema_relation }} limit {{ max_results_per_iter }}
   {%- endset -%}
 
   {%- set result = run_query(sql) -%}
-  {% set maximum = 10000 %}
-  {% if (result | length) >= maximum %}
-    {% set msg %}
-      Too many objects in schema  {{ schema_relation }}! dbt can only get
-      information about schemas with fewer than {{ maximum }} objects.
-    {% endset %}
-    {% do exceptions.raise_compiler_error(msg) %}
+  {%- set n = (result | length) -%}
+  {%- set counts_array = [n] -%}
+  {%- set paginated_results_array = [] -%}
+
+  {% if n >= max_results_per_iter %}
```

Comment:
this is such a long `if`, at line 224 (I think?) it be great to pull this into its own function with a descriptive name like `get_paginated_relations_array` (_vel sim._) and just have a function call. 

If for whatever reason that's not possible, you could also leave a comment online 224. My first suggestion is much preferred though

Reply:
Oh, that's right Ginger is a pain about this. I've got a work around for you:

```
{% set ns = namespace(sum=0) %}
{% for value in values %}
  {%- set ns.sum = ns.sum + value %}
{% endfor %}
The sum of the values is: {{ ns.sum }}
```

Reply:
super helpful, thanks